### ISSUE TITLE
Can't add a card to an account

### ISSUE THREAD

**Issue by lefta** [stripe/stripe-go#478]
When adding a card to a client, the Stripe API returns a 400 error with a `Received unknown parameter: source` message.

Example code:

```go
params := &stripe.CardParams{
	Account: "acct_...",
	Token:   "tok_mastercard_debit_transferFailure",
}
card.New(params)
```

The problem seems to lie in `AppendToAsCardSourceOrExternalAccount`. The function adds the source field even when params contains an account.

As a workaround, I was able to solve this issue by reimplementing the relevant parts of `card.New()`:

```go
c := card.Client{stripe.GetBackend(stripe.APIBackend), stripe.Key}
body := &form.Values{}
body.Add(form.FormatKey([]string{"external_account"}), params.Token)
card := &stripe.Card{}
err := c.B.Call("POST", fmt.Sprintf("/accounts/%v/external_accounts", params.Account), c.Key, body, &params.Params, card)
```

Go version: 1.9.1
Stripe Go library version: 28.3.0

**Comment by remi-stripe** [comment]
Hey @lefta. Thanks a lot for the detailed report! I pushed a fix in https://github.com/stripe/stripe-go/pull/479 which solves the issue. Can you confirm this has no side effect in your current project?

**Comment by brandur-stripe** [comment]
Hi there, we've released a fix for this as part of 28.3.2. Thanks for reporting and sorry about the trouble!
